File: 82.cpp



class Solution {
public:
	/**
	 * @param A: Array of integers.
	 * return: The single number.
	 */
    int singleNumber(vector<int> &A) {
        // write your code here
        int result = 0,n=A.size();
	    for (int i = 0; i<n; i++)
	    {
	        result ^=A[i];
	    }
	    return result;
    }
};

File: 82.py
class Solution:
    # @param A, a list of integer
    # @return an integer
    def singleNumber(self, A):
        A.sort()
        if len(A)==0:
        	return 0
        	
        for i in range(1, len(A), 2):
            if A[i] != A[i-1]: 
                return A[i-1]
        return A[-1] 


